stylelint-config/rules/use-defined-css-var-when-available/props-to-allowed-vars.js

```diff
@@ -0,0 +1,129 @@
+  "--font-size-general-label",
+  "--font-size-paragraph",
+  "--font-size-subtext",
+];
+
+const FONT_WEIGHTS = ["--font-weight-very-light", "--font-weight-light", "--font-weight-normal", "--font-weight-bold"];
+
+const FONT_LINE_HEIGHTS = [
+  "--font-line-height-10",
+  // since "--font-line-height-20", "--font-line-height-30" and "--font-line-height-40" have the same value, we need to pick only one of them as "valid value", to allow autofix
+  // "--font-line-height-20",
+  // "--font-line-height-30",
+  "--font-line-height-40",
+  "--font-line-height-50",
+  "--font-line-height-60",
+  "--font-line-height-h1",
+  "--font-line-height-h2",
+  "--font-line-height-h3",
+  "--font-line-height-h4",
+  "--font-line-height-h5",
+  "--font-line-height-general-label",
+  "--font-line-height-paragraph",
+  "--font-line-height-subtext",
+];
+
+const SPACING_PROPS = [
+  "padding",
+  "padding-top",
+  "padding-bottom",
+  "padding-left",
+  "padding-right",
+  "padding-inline",
+  "padding-inline-end",
+  "padding-inline-start",
+  "padding-block",
+  "padding-block-end",
+  "padding-block-start",
+
+  "margin",
+  "margin-top",
+  "margin-bottom",
+  "margin-left",
+  "margin-right",
+  "margin-inline",
+  "margin-inline-end",
+  "margin-inline-start",
+  "margin-block",
+  "margin-block-end",
+  "margin-block-start",
+
+  "top",
+  "left",
+  "bottom",
+  "right",
+
+  "inset",
+  "inset-end",
+  "inset-start",
+];
+
+const BORDER_RADIUSES_PROPS = [
+  "border-radius",
+  "border-top-left-radius",
+  "border-top-right-radius",
+  "border-bottom-left-radius",
+  "border-bottom-right-radius",
+  "border-start-start-radius",
+  "border-start-end-radius",
+  "border-end-start-radius",
+  "border-end-end-radius",
+];
+
+const TIMING_FUNCTION_PROPS = ["transition", "transition-timing", "animation", "animation-timing-function"];
+
+function mapPropsToAllowedVars(propNames, allowedVars) {
+  allowedVars = Array.isArray(allowedVars) ? allowedVars : [allowedVars];
+  propNames = Array.isArray(propNames) ? propNames : [propNames];
+
+  return propNames.reduce((result, propName) => ({ ...result, [propName]: allowedVars }), {});
+}
+
+// List the CSS props we want to lint, and map each one to the values we would prefer to use.
+// For example, let's say that the key "border-radius" is mapped to --border-radius-small and --border-radius-medium.
+// This means that if --border-radius-small or --border-radius-medium can be used while linting a rule with the property "border-radius", we will show an error
+
+const PROPS_TO_ALLOWED_VARS = {
+  ...mapPropsToAllowedVars(SPACING_PROPS, SPACINGS),
+  ...mapPropsToAllowedVars(BORDER_RADIUSES_PROPS, BORDER_RADIUSES),
+  ...mapPropsToAllowedVars("line-height", FONT_LINE_HEIGHTS),
+  ...mapPropsToAllowedVars("font-weight", FONT_WEIGHTS),
+  ...mapPropsToAllowedVars("font-size", FONT_SIZES),
+  ...mapPropsToAllowedVars("box-shadow", BOX_SHADOWS),
+  ...mapPropsToAllowedVars(TIMING_FUNCTION_PROPS, "--expand-animation-timing"),
```

Comment:
the way this rule works is that IF a CSS rule has a prop that's matching one of `TIMING_FUNCTION_PROPS`, and it has the value of `--expand-animation-timing`, then there will be an error saying "Please use `--expand-animation-timing` instead of `bla bla`". 
We'll might have issues where the value has spaces in it, but I think we should be ok for now. 😃 
Does it make sense?